x/smart/stream_dialer.go

```diff
@@ -185,6 +186,22 @@ func (f *StrategyFinder) dnsConfigToResolver(dnsConfig []dnsEntryConfig) ([]*sma
 	return rts, nil
 }
 
+// create a StreamDialer from a configURL, for example "ss://..."
+func getStreamDialer(ctx context.Context, configURL string) (transport.StreamDialer, error) {
+	config, err := configurl.ParseConfig(configURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	providers := configurl.NewDefaultProviders()
```

Comment:
Can you cache the providers so we don't need to create them for every entry and reduce allocations? See how we reuse the `configModule` in `findTLS`

Reply:
done